### PR TITLE
Fix ruby version errors with eventmachine

### PR DESCRIPTION
**Pull request by thisisdano**
I was finding installation issues similar to https://github.com/uswds/uswds-site/issues/1745 (M2 Mac / macOS Monterey) but since I'm using asdf and not RVM, I wasn't able to follow the fix instructions from that issue exactly, and the adapted fix I tried did not work. 

This pull request updates the Ruby version to 3.1.2 and supports the `asdf` version manager with a `.tool-versions` config. This helped solve my problems getting a new local version of the website running on an M2 Mac running Monterey. 

We'll still need to test this a bit more broadly, with all team machines, before merging.

- Are we all able to use Ruby 3.1.2?
- Does the `bundle install` work using 3.1.2 and your Ruby version manager of choice (ie RVM)?

When running USWDS Site locally:

1. Install Ruby 3.1.2 using your Ruby version manager
1. Use Ruby 3.1.2 in the project
1. Delete any existing `Gemfile.lock`
1. Run `gem install bundler:2.2.0` to use a version of bundler that matches our `.bundler-version`
1. Run `bundle install`

### CODE REVIEW

**Comment by mitchmoccia** [comment]
@thisisdano I did the following:
- pulled your branch
- installed Ruby 3.1.2 with RVM `rvm install ruby-3.1.2`
- `rvm list` to confirm 3.1.2 was in use
- `bundle install` successfully
- `npm install` successfully
- successfully stood up the site locally!

So it appears your update works with RVM and Ruby 3.1.2

**Review by mitchmoccia — APPROVED** [review]
@thisisdano I did the following:

pulled your branch
installed Ruby 3.1.2 with RVM rvm install ruby-3.1.2
rvm list to confirm 3.1.2 was in use
bundle install successfully
npm install successfully
successfully stood up the site locally!
So it appears your update works with RVM and Ruby 3.1.2

**Comment by mejiaj** [comment]
macOS Monterey v12.6

- [x] Ran `rvm install ruby-3.1.2` successfully
- [ ] Bundle install returns error below

```bash
An error occurred while installing eventmachine (1.2.7), and Bundler cannot continue.

In Gemfile:
  jekyll-include-cache was resolved to 0.2.1, which depends on
    jekyll was resolved to 4.2.2, which depends on
      em-websocket was resolved to 0.5.3, which depends on
        eventmachine
```
 
<details>
  <summary>Full error</summary>

  ```
  uswds-site on  dw-fix-color-tokens-nav [!1] via  v16.17.0 via 💎 v3.1.2 
➜ bundle install
Fetching gem metadata from https://rubygems.org/.........
Using public_suffix 4.0.7
Using bundler 2.3.17
Using colorator 1.1.0
Using concurrent-ruby 1.1.10
Using diff-lcs 1.5.0
Using http_parser.rb 0.8.0
Using ffi 1.15.5
Using forwardable-extended 2.6.0
Using mercenary 0.4.0
Using mini_portile2 2.8.0
Using racc 1.6.0
Using parallel 1.22.1
Using rainbow 3.1.1
Using yell 2.2.2
Using zeitwerk 2.6.0
Using rb-fsevent 0.11.1
Using rexml 3.2.5
Using rouge 3.30.0
Using safe_yaml 1.0.5
Using unicode-display_width 1.8.0
Using json 2.6.2
Using rspec-support 3.11.0
Using webrick 1.7.0
Using nokogiri 1.13.8
Using liquid 4.0.3
Using kramdown 2.4.0
Using terminal-table 2.0.0
Using rspec-core 3.11.0
Using i18n 1.12.0
Using kramdown-parser-gfm 1.1.0
Using liquid-c 4.0.0
Using addressable 2.8.0
Using rspec-expectations 3.11.0
Using pathutil 0.16.2
Using ethon 0.15.0
Using sassc 2.4.0
Using rb-inotify 0.10.1
Using typhoeus 1.4.0
Fetching jekyll-sass-converter 2.2.0
Using listen 3.7.1
Using html-proofer 4.3.2
Using jekyll-watch 2.2.1
Installing eventmachine 1.2.7 with native extensions
Installing jekyll-sass-converter 2.2.0
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/eventmachine-1.2.7/ext
/Users/jmejia-a/.rvm/rubies/ruby-3.1.2/bin/ruby -I /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0 -r
./siteconf20221011-32872-ilvroh.rb extconf.rb
checking for -lcrypto... yes
checking for -lssl... yes
checking for openssl/ssl.h... yes
checking for openssl/err.h... yes
checking for rb_trap_immediate in ruby.h,rubysig.h... no
checking for rb_thread_blocking_region()... no
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
checking for rb_thread_fd_select()... yes
checking for rb_fdset_t in ruby/intern.h... yes
checking for rb_wait_for_single_fd()... yes
checking for rb_enable_interrupt()... no
checking for rb_time_new()... yes
checking for inotify_init() in sys/inotify.h... no
checking for __NR_inotify_init in sys/syscall.h... no
checking for writev() in sys/uio.h... yes
checking for pipe2() in unistd.h... no
checking for accept4() in sys/socket.h... no
checking for SOCK_CLOEXEC in sys/socket.h... no
checking for sys/event.h... yes
checking for sys/queue.h... yes
checking for clock_gettime()... yes
checking for CLOCK_MONOTONIC_RAW in time.h... yes
checking for CLOCK_MONOTONIC in time.h... yes
CXXFLAGS=-fdeclspec -Wall -Wextra -Wno-deprecated-declarations -Wno-ignored-qualifiers -Wno-unused-result -Wno-address
creating Makefile

current directory: /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/eventmachine-1.2.7/ext
make DESTDIR\= clean

current directory: /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/eventmachine-1.2.7/ext
make DESTDIR\=
compiling binder.cpp
In file included from binder.cpp:20:
./project.h:119:10: fatal error: 'openssl/ssl.h' file not found
#include <openssl/ssl.h>
         ^~~~~~~~~~~~~~~
1 error generated.
make: *** [binder.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/eventmachine-1.2.7 for inspection.
Results logged to /Users/jmejia-a/.rvm/gems/ruby-3.1.2/extensions/arm64-darwin-21/3.1.0/eventmachine-1.2.7/gem_make.out

  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:95:in `run'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:44:in `block in make'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:36:in `each'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:36:in `make'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/ext_conf_builder.rb:63:in `block in build'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/tempfile.rb:317:in `open'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/ext_conf_builder.rb:26:in `build'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:161:in `build_extension'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:195:in `block in build_extensions'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `each'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `build_extensions'
  /Users/jmejia-a/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/installer.rb:853:in `build_extensions'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/rubygems_gem_installer.rb:71:in `build_extensions'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/rubygems_gem_installer.rb:28:in `install'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/source/rubygems.rb:209:in `install'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/installer/gem_installer.rb:54:in `install'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
/Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/installer/parallel_installer.rb:177:in `block in
worker_pool'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/worker.rb:62:in `apply_func'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/worker.rb:57:in `block in process_queue'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/worker.rb:54:in `loop'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/worker.rb:54:in `process_queue'
  /Users/jmejia-a/.rvm/gems/ruby-3.1.2/gems/bundler-2.3.17/lib/bundler/worker.rb:91:in `block (2 levels) in create_threads'

An error occurred while installing eventmachine (1.2.7), and Bundler cannot continue.

In Gemfile:
  jekyll-include-cache was resolved to 0.2.1, which depends on
    jekyll was resolved to 4.2.2, which depends on
      em-websocket was resolved to 0.5.3, which depends on
        eventmachine
  ```

</details>

**Comment by mejiaj** [comment]
I was able to get my local site working again with the command below.

`gem install eventmachine -- --with-openssl-dir=$(brew --prefix openssl@1.1)`

**Review by mahoneycm — COMMENTED** [review]
I after installing `ruby-3.1.2` I ran into the same `bundle install` error as @mejiaj. I tried the fix mentioned but whenever I run `npm run start` it gets stuck on `watchSass` and never finishes

**Comment by amyleadem** [comment]
I was able to run the following commands successfully:

Setup: MacOS Monterey v12.6/ Intel Core i7
- Ran `rvm install "ruby-3.1.2"`
- Ran `rvm use 3.1.2`
- Confirmed use of v3.1.2
- Removed  files: .jekyll*, package-lock.json, node_modules, Gemfile.lock
- Ran `bundle install`
- Ran `npm install`
- Ran `npm start`

**Review by amyleadem — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-gem 'jekyll', '>= 4.2.0'
+gem 'jekyll', '>= 4.3.1'
 gem 'json', '>= 2.3.0'
 gem 'jekyll-redirect-from', '>= 0.15.0'
 gem 'jekyll-sitemap', '>= 1.4.0'
 gem 'jekyll-include-cache'
 gem 'liquid-c'
-gem 'html-proofer'
+gem 'html-proofer', '= 4.4.3'
```
Comment (thisisdano) [inline Gemfile:9]: `html-proofer` >= 5.0.0 breaks our workflow

File: Gemfile [hunk 2]
```diff
@@ -1,12 +1,12 @@
 source 'https://rubygems.org'
 
-gem 'jekyll', '>= 4.2.0'
+gem 'jekyll', '>= 4.3.1'
```
Comment (thisisdano) [inline Gemfile:3]: Most recent Jekyll

File: .gitignore [hunk 3]
```diff
@@ -16,3 +16,5 @@ pa11y-results.json
 # Ignore VS code directories
 *.code-workspace
 .vscode/
+
+Gemfile.lock
```
Comment (thisisdano) [inline .gitignore:20]: The lockfile is encoding ARM-specific dependencies that could be breaking some installations. If we ignore it, we may have more success installing across devices — at the expense of standardization, but maybe that's worth it

**Pull request merged**